pkg/services/librarypanels/librarypanels.go

```diff
@@ -179,8 +180,24 @@ func (lps *LibraryPanelService) ConnectLibraryPanelsForDashboard(c *models.ReqCo
 		if len(uid) == 0 {
 			return errLibraryPanelHeaderUIDMissing
 		}
-		libraryPanels = append(libraryPanels, uid)
+		_, exists := libraryPanels[uid]
+		if !exists {
+			libraryPanels[uid] = uid
+		}
+	}
+
+	elementUIDs := make([]string, 0, len(libraryPanels))
+	for libraryPanel := range libraryPanels {
+		elementUIDs = append(elementUIDs, libraryPanel)
```

Comment:
While fixing the issue, I realized that we were potentially calling connect several times for the same UID because we weren't using a map. I added this little snippet to handle that.